### PR TITLE
mining keys to open tendrills crates buyable with miner points

### PR DESCRIPTION
**Pull request by bloons3**
Ports https://github.com/tgstation/tgstation/pull/56267
## About The Pull Request

image

image

And ashwalkers have it!

image

image

## Changelog
:cl: Tlaltecuhtli 
balance: tendril crates need keys to open which you can spend mining points to get
balance: ashwalkers can craft em with bones cause they got no points
/:cl:

### CODE REVIEW

**Comment by Minelol76** [comment]
Real question: Is key crafting Exclusive to As(s)hwalkers?

**Comment by bloons3** [comment]
> Real question: Is key crafting Exclusive to As(s)hwalkers?

Yes. Ashwalkers craft keys from bones, miners buy keys.

**Comment by super12pl** [comment]
Whats the point of this PR being made after we already brutalised miner loot?

**Comment by OceanFish1** [comment]
> Whats the point of this PR being made after we already brutalised miner loot?

more nerf = less buff = better, thats how coding works

**Comment by ToasterBan** [comment]
why 777 points? that's a dumb number just set it to like 500

**Comment by bloons3** [comment]
> why 777 points? that's a dumb number just set it to like 500

Highly spiritual Angle Number 777 is the Sign of getting Divine Guidance. That indicates Its time to get rewards for your efforts.
Angel numbers can mean different things. However, if you are seeing the angel number often and often then you should be happy. That is because the angel number means only positive things. As you are seeing the number that means something good is waiting for you.

❤  5 Unusual Facts About Angel Number 777 extension  ❤

1. Angel Number 777 indicating the path of following your guardian angels
2. Repeatedly Seeing the Angel Number 777 means you’re Going to right way to achieving you Success. 
3. It also Indicating negative connotation Due to gambling and thats help to know as positive number
4. Angel Number 777 also helps to know by come frequently when you are enjoying a string of luck
5. When Number comes with a promise of abundance that means Everything in abundance wanted in your life.

File: code/modules/mining/machine_vending.dm [hunk 1]
```diff
@@ -12,9 +12,12 @@
 	var/icon_deny = "mining-deny"
 	var/obj/item/card/id/inserted_id
 	var/list/prize_list = list( //if you add something to this, please, for the love of god, sort it by price/type. use tabs and not spaces.
-		new /datum/data/mining_equipment("30 Marker Beacons",			/obj/item/stack/marker_beacon/thirty,								100),
+		new /datum/data/mining_equipment("1 Marker Beacon",				/obj/item/stack/marker_beacon,										5),
+		new /datum/data/mining_equipment("10 Marker Beacons",			/obj/item/stack/marker_beacon/ten,									500),
+		new /datum/data/mining_equipment("30 Marker Beacons",			/obj/item/stack/marker_beacon/thirty,								150),
 		new /datum/data/mining_equipment("Shelter Capsule",				/obj/item/survivalcapsule,											400),
 		new /datum/data/mining_equipment("Regen. Core Stabilizer",		/obj/item/hivelordstabilizer,										400),
+		new /datum/data/mining_equipment("Skeleton Key",				/obj/item/skeleton_key,												777),
```
Comment (super12pl) [inline code/modules/mining/machine_vending.dm]: For the love of god

**Comment by Effemu** [comment]
> 
> 
> Whats the point of this PR being made after we already brutalised miner loot?

yeah wtf bloons

**Comment by aidan-bowman** [comment]
if you're gonna do this then add back in some of the shit that ruko yeeted

**Comment by yyzsong** [comment]
Didnt someone already try this?

File: code/modules/mining/machine_vending.dm [hunk 2]
```diff
@@ -12,9 +12,12 @@
 	var/icon_deny = "mining-deny"
 	var/obj/item/card/id/inserted_id
 	var/list/prize_list = list( //if you add something to this, please, for the love of god, sort it by price/type. use tabs and not spaces.
-		new /datum/data/mining_equipment("30 Marker Beacons",			/obj/item/stack/marker_beacon/thirty,								100),
+		new /datum/data/mining_equipment("1 Marker Beacon",				/obj/item/stack/marker_beacon,										5),
+		new /datum/data/mining_equipment("10 Marker Beacons",			/obj/item/stack/marker_beacon/ten,									500),
```
Comment (yyzsong) [inline code/modules/mining/machine_vending.dm:16]: undocu ment ed change!!!

**Comment by Sinestia** [comment]
> if you're gonna do this then add back in some of the shit that ruko yeeted

You could totally do this on your own

Just be sure to make necessarily tweaks to get them actually allowed in.

Like I think /tg/ adequately fixed the cursed katana by making it give you a permanent stalker trauma.

**Comment by Sinestia** [comment]
> why 777 points? that's a dumb number just set it to like 500

If anything it's a bit too cheap but w.e


777 has soul. You are Soulless.

**Comment by utilbird** [comment]
\> Mining keys PR opened on bee (#3342)
\> Doesn't get merged
\> TG <strike>steals</strike> makes their own refined Key PR just a few days after
\> We port the TG PR

We've broke the system

**Comment by That0nePerson** [comment]
> Didnt someone already try this?

I did, yes

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 3]
```diff
@@ -11,16 +11,28 @@
 	door_anim_time = 0
 
 /obj/structure/closet/crate/necropolis/tendril
-	desc = "It's watching you suspiciously."
+	desc = "It's watching you suspiciously. You need a skeleton key to open it."
```
Comment (PowerfulBacon) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Set the you need a key part in the examine, so when you examine it after its unlocked it doesn't show that.

File: code/datums/components/crafting/recipes.dm [hunk 4]
```diff
@@ -780,6 +780,13 @@
 	result = /obj/structure/bonfire
 	category = CAT_PRIMAL
 
+/datum/crafting_recipe/skeleton_key
+	name = "Skeleton Key"
+	time = 30
+	reqs = list(/obj/item/stack/sheet/bone = 5)
+	result = /obj/item/skeleton_key
+	always_available = FALSE
+	category = CAT_PRIMAL
```
Comment (PowerfulBacon) [inline code/datums/components/crafting/recipes.dm:789]: add a space after 🥺

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 5]
```diff
@@ -11,16 +11,28 @@
...
+	/// var to check if it got opened by a key
+	var/spawned_loot = FALSE
 
-/obj/structure/closet/crate/necropolis/tendril/PopulateContents()
+/obj/structure/closet/crate/necropolis/tendril/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/try_spawn_loot)
+
+/obj/structure/closet/crate/necropolis/tendril/proc/try_spawn_loot(datum/source, obj/item/item, mob/user, params) ///proc that handles key checking and generating loot
+	SIGNAL_HANDLER
```
Comment (PowerfulBacon) [inline code/modules/mining/lavaland/necropolis_chests.dm:25]: I mean its fine but wouldnt it just be better to use the attackby verb directly?

File: code/modules/mining/machine_vending.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/modules/mining/machine_vending.dm]: 777 is a weird and pretty low number.

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (PowerfulBacon) [reply code/datums/components/crafting/recipes.dm:789]: nl*

**Comment by snek-with-tophat** [comment]
fuck hermit, beach and doctor spawners i guess

**Comment by Sinestia** [comment]
> fuck hermit, beach and doctor spawners i guess

I dunno if anyone would be opposed to those being removed, all people use them for is extra lives. Not actually because there's anything interesting about them like golems/lifegivers/syndicate/ashwalkers.

**Comment by bloons3** [comment]
> > fuck hermit, beach and doctor spawners i guess
> 
> I dunno if anyone would be opposed to those being removed, all people use them for is extra lives. Not actually because there's anything interesting about them like golems/lifegivers/syndicate/ashwalkers.

~~I ded, I want be convert by cult~~

Every other spawner can enter the mining station without being killed on sight. Ashwalkers are also the Lavaland natives, so it makes sense that they can make the keys required to obtain loot from chests.

**Comment by Rukofamicom** [comment]
I think these should cost 1500 or so.

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 3 above)
Comment (bloons3) [reply code/modules/mining/lavaland/necropolis_chests.dm]: image

image

image

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 5 above)
Comment (bloons3) [reply code/modules/mining/lavaland/necropolis_chests.dm:25]: Signals

File: code/datums/components/crafting/recipes.dm (hunk 4 above)
Comment (bloons3) [reply code/datums/components/crafting/recipes.dm:789]: added

File: code/modules/mining/machine_vending.dm (hunk 1 above)
Comment (bloons3) [reply code/modules/mining/machine_vending.dm]: Cost is arbitrary, this is just to get miners to drop ore off and hopefully decrease the probability that a single miner manages to loot every chest (leaving no other chests for other miners).

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 6]
```diff
@@ -58,6 +70,21 @@
...
+
+/obj/structure/closet/crate/necropolis/tendril/can_open(mob/living/user, force = FALSE)
+	if(!spawned_loot)
+		return FALSE
+	return ..()
+
+/obj/structure/closet/crate/necropolis/tendril/examine(mob/user)
+	. = ..()
+	if(!spawned_loot)
+		. += " You need a skeleton key to open it."
```
Comment (utilbird) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
		. += "<span class='notice'>You need a skeleton key to open it.</span>"
```

File: code/modules/mining/machine_vending.dm (hunk 1 above)
Comment (utilbird) [inline code/modules/mining/machine_vending.dm]: make it consistent with the cost of other items. 1000 would be good

Comment (bloons3) [reply code/modules/mining/machine_vending.dm]: Amount is relatively arbitrary, but somewhere between 500 and 1000 makes it a requirement but not something too annoying to do

Comment (utilbird) [reply code/modules/mining/machine_vending.dm]: @bloons3 What I mean is, make it a multiple of 100 or at least a multiple of 50

Comment (ike709) [reply code/modules/mining/machine_vending.dm]: I concur, it should be a multiple of 50 or 100. No funny numbers.

Comment (bloons3) [reply code/modules/mining/machine_vending.dm]: 750

**Review by ike709 — DISMISSED** [review]

**Review by ike709 — APPROVED** [review]

File: code/modules/mining/machine_vending.dm (hunk 2 above)
Comment (PowerfulBacon) [inline code/modules/mining/machine_vending.dm:16]: wait why are 10 marker beacons more expensive than 30?

**Pull request merged**